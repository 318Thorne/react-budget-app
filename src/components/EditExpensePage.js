import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {editExpense, startRemoveExpense} from '../actions/expenses';


export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
            this.props.onSubmit(this.props.expense.id, expense);
            this.props.history.push('/');
        };
    onClick = () => {
            const id = this.props.expense.id
            this.props.onClick(id);
            this.props.history.push('/');
        };

    render(){
        return (
            <div>
                <ExpenseForm 
                    expense = {this.props.expense}
                    onSubmit={this.onSubmit}                  
                />
                <button onClick={this.onClick}>Remove item</button>
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        onSubmit: (id, expense) => dispatch(editExpense(id, expense)),
        onClick: (id) => dispatch(startRemoveExpense({id}))       
    }
}

const mapStateToProps = (state, props) => {
    return{
        expense: state.expenses.find((expense) => (expense.id === props.match.params.id) )
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);