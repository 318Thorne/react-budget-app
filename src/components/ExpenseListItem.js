import React from 'react';
import {Link} from 'react-router-dom';


export const ExpenseListItem = ({id , description, amount, createdAt}) => (
    <div>
        <p>{description}</p>
        <p>{amount}</p>
        <p>{createdAt}</p>
        <Link to={`/edit/${id}`}>edit</Link>
        <hr/>
    </div>
);

// const mapStateToProps = (state) => {
//     return {
//         expenses: state.expenses
//     }
// }

export default ExpenseListItem