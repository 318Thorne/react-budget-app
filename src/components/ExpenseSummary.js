import React from 'react';
import { connect } from 'react-redux';
import ExpenseTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses'
import numeral from 'numeral'

export const ExpenseSummary = ({expensesCount, expensesTotal}) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses'
    return (
        <div>            
            <p>Viewing {expensesCount} {expenseWord} totalling {numeral(expensesTotal / 100).format('$0,0.00')} </p>
        </div>
    )
}

const mapStateToProps = (state) => {
    const expenses = selectExpenses(state.expenses, state.filters)
    return {
        expensesCount: expenses.length,
        expensesTotal: ExpenseTotal(expenses)
    }
}

export default connect(mapStateToProps)(ExpenseSummary);