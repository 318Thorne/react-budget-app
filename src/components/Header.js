import React from 'react';
import {NavLink} from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <ul>
            <li><NavLink to="/" activeClassName="isActive" exact={true}>Home</NavLink></li>
            <li><NavLink to="/create" activeClassName="isActive">Add</NavLink></li>
            <li><NavLink to="/help" activeClassName="isActive" >Help</NavLink></li>
        </ul>
    </header>
);
export default Header;