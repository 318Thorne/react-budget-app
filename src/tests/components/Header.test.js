import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Header } from '../../components/Header';

test('should render Header correctly', () => {
    const wrapper = shallow(<Header startLogout={() => {} }/>)
    expect(wrapper).toMatchSnapshot();
})

test('should call start logout on button click', () => {
    const onLogoutSpy = jest.fn();
    const wrapper = shallow(<Header startLogout={onLogoutSpy}/>)
    wrapper.find('#logout').simulate('click')
    expect(onLogoutSpy).toHaveBeenCalled();
})
