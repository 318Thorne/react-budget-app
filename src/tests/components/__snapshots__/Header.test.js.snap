// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Expensify
        </h1>,
        <ul>
          <li>
            <NavLink
              activeClassName="isActive"
              aria-current="page"
              exact={true}
              to="/"
            >
              Home
            </NavLink>
          </li>
          <li>
            <NavLink
              activeClassName="isActive"
              aria-current="page"
              to="/create"
            >
              Add
            </NavLink>
          </li>
          <li>
            <NavLink
              activeClassName="isActive"
              aria-current="page"
              to="/help"
            >
              Help
            </NavLink>
          </li>
        </ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Expensify",
        },
        "ref": null,
        "rendered": "Expensify",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <NavLink
                activeClassName="isActive"
                aria-current="page"
                exact={true}
                to="/"
              >
                Home
              </NavLink>
            </li>,
            <li>
              <NavLink
                activeClassName="isActive"
                aria-current="page"
                to="/create"
              >
                Add
              </NavLink>
            </li>,
            <li>
              <NavLink
                activeClassName="isActive"
                aria-current="page"
                to="/help"
              >
                Help
              </NavLink>
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="isActive"
                aria-current="page"
                exact={true}
                to="/"
              >
                Home
              </NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "isActive",
                "aria-current": "page",
                "children": "Home",
                "exact": true,
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="isActive"
                aria-current="page"
                to="/create"
              >
                Add
              </NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "isActive",
                "aria-current": "page",
                "children": "Add",
                "to": "/create",
              },
              "ref": null,
              "rendered": "Add",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="isActive"
                aria-current="page"
                to="/help"
              >
                Help
              </NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "isActive",
                "aria-current": "page",
                "children": "Help",
                "to": "/help",
              },
              "ref": null,
              "rendered": "Help",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Expensify
          </h1>,
          <ul>
            <li>
              <NavLink
                activeClassName="isActive"
                aria-current="page"
                exact={true}
                to="/"
              >
                Home
              </NavLink>
            </li>
            <li>
              <NavLink
                activeClassName="isActive"
                aria-current="page"
                to="/create"
              >
                Add
              </NavLink>
            </li>
            <li>
              <NavLink
                activeClassName="isActive"
                aria-current="page"
                to="/help"
              >
                Help
              </NavLink>
            </li>
          </ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Expensify",
          },
          "ref": null,
          "rendered": "Expensify",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <NavLink
                  activeClassName="isActive"
                  aria-current="page"
                  exact={true}
                  to="/"
                >
                  Home
                </NavLink>
              </li>,
              <li>
                <NavLink
                  activeClassName="isActive"
                  aria-current="page"
                  to="/create"
                >
                  Add
                </NavLink>
              </li>,
              <li>
                <NavLink
                  activeClassName="isActive"
                  aria-current="page"
                  to="/help"
                >
                  Help
                </NavLink>
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="isActive"
                  aria-current="page"
                  exact={true}
                  to="/"
                >
                  Home
                </NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "isActive",
                  "aria-current": "page",
                  "children": "Home",
                  "exact": true,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Home",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="isActive"
                  aria-current="page"
                  to="/create"
                >
                  Add
                </NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "isActive",
                  "aria-current": "page",
                  "children": "Add",
                  "to": "/create",
                },
                "ref": null,
                "rendered": "Add",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="isActive"
                  aria-current="page"
                  to="/help"
                >
                  Help
                </NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "isActive",
                  "aria-current": "page",
                  "children": "Help",
                  "to": "/help",
                },
                "ref": null,
                "rendered": "Help",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
